lib/fhir/query-utils.js

```diff
@@ -210,6 +210,22 @@ module.exports = (mongo) => {
       return clause
     },
 
+    tokenToConditionalElemMatch: (fieldToMatch, token, condition) => {
+      const conditionSplitArray = condition.replace(/'/g, '').split('=')
+      const conditionPath = conditionSplitArray[0]
+      const conditionValue = conditionSplitArray[1]
+
+      const clause = {}
+      clause[fieldToMatch] = {
+        $elemMatch: {
+          value: token
+        }
+      }
+      clause[fieldToMatch]['$elemMatch'][conditionPath.replace('.value', '')] = conditionValue
```

Comment:
I do think we should remove the `.value` in the gen script. It should be a fairly quick change. Perhaps use the regex to extract the condition then replace the .value as you do here.